Microsoft.Toolkit.Uwp.UI/Extensions/NavigationView/NavigationViewExtensions.cs

```diff
@@ -0,0 +1,181 @@
+        /// <returns>The selected index.</returns>
+        public static int GetSelectedIndex(NavigationView obj)
+        {
+            return (int)obj.GetValue(SelectedIndexProperty);
+        }
+
+        /// <summary>
+        /// Sets the index of the selected <see cref="NavigationViewItem"/>.
+        /// </summary>
+        /// <param name="obj">The <see cref="Windows.UI.Xaml.Controls.NavigationView"/>.</param>
+        /// <param name="value">The index to select.</param>
+        public static void SetSelectedIndex(NavigationView obj, int value)
+        {
+            obj.SetValue(SelectedIndexProperty, value);
+        }
+
+        /// <summary>
+        /// Attached <see cref="DependencyProperty"/> for binding the selected index of a <see cref="NavigationView"/>.
+        /// </summary>
+        public static readonly DependencyProperty SelectedIndexProperty = DependencyProperty.RegisterAttached("SelectedIndex", typeof(int), typeof(NavigationViewExtensions), new PropertyMetadata(-1, OnSelectedIndexChanged));
+
+        /// <summary>
+        /// Gets the behavior to collapse the content when clicking the already selected <see cref="NavigationViewItem"/>.
+        /// </summary>
+        /// <param name="obj">The <see cref="Windows.UI.Xaml.Controls.NavigationView"/>.</param>
+        /// <returns>True if the feature is on.</returns>
+        public static bool GetCollapseOnClick(NavigationView obj)
+        {
+            return (bool)obj.GetValue(CollapseOnClickProperty);
+        }
+
+        /// <summary>
+        /// Sets the behavior to collapse the content when clicking the already selected <see cref="NavigationViewItem"/>.
+        /// </summary>
+        /// <param name="obj">The <see cref="Windows.UI.Xaml.Controls.NavigationView"/>.</param>
+        /// <param name="value">True to turn on this feature.</param>
+        public static void SetCollapseOnClick(NavigationView obj, bool value)
+        {
+            obj.SetValue(CollapseOnClickProperty, value);
+        }
+
+        /// <summary>
+        /// Attached <see cref="DependencyProperty"/> for enabling the behavior to collapse the <see cref="NavigationView"/> content when the same selected item is invoked again (click or tap).
+        /// </summary>
+        public static readonly DependencyProperty CollapseOnClickProperty =
+            DependencyProperty.RegisterAttached("CollapseOnClick", typeof(bool), typeof(NavigationViewExtensions), new PropertyMetadata(false, OnCollapseOnClickChanged));
+
+        private static void OnCollapseOnClickChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var navview = d as Windows.UI.Xaml.Controls.NavigationView;
+
+            if (navview == null)
+            {
+                return;
+            }
+
+            navview.ItemInvoked -= Navview_ItemInvoked;
+
+            if (e.NewValue as bool? == true)
```

Comment:
I don't see a scenario where e.NewValue could be null. a direct cast is efficient.